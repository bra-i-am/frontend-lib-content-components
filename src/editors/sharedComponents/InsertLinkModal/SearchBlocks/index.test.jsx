import React from 'react';
import { IntlProvider } from '@edx/frontend-platform/i18n';
import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

import SearchBlocks from '.';

Enzyme.configure({ adapter: new Adapter() });

const mockBlocks = {
  'block-key': {
    id: 'block-key',
    blockId: 'edx_block-1',
    lmsWebUrl: 'http://localhost/weburl',
    legacyWebUrl: 'http://localhost/legacy',
    studentViewUrl: 'http://localhost/studentview',
    type: 'sequential',
    displayName: 'Any display name',
    path: 'Any display name',
    children: ['block-children-1', 'block-children-2'],
  },
  'block-children-1': {
    id: 'block-children-1',
    blockId: 'edx_block-1',
    lmsWebUrl: 'http://localhost/weburl',
    legacyWebUrl: 'http://localhost/legacy',
    studentViewUrl: 'http://localhost/studentview',
    type: 'sequential',
    displayName: 'Block children 1',
    path: 'Any display name / Block children 1',
  },
  'block-children-2': {
    id: 'block-children-2',
    blockId: 'edx_block-2',
    lmsWebUrl: 'http://localhost/weburl',
    legacyWebUrl: 'http://localhost/legacy',
    studentViewUrl: 'http://localhost/studentview',
    type: 'sequential',
    displayName: 'Block children 2',
    path: 'Any display name / Block children 2',
  },
};

describe('SearchBlocks Component', () => {
  // eslint-disable-next-line react/prop-types
  const IntlProviderWrapper = ({ children }) => (
    <IntlProvider locale="en" messages={{}}>
      {children}
    </IntlProvider>
  );

  const onSearchFilterMock = jest.fn();
  const onBlockSelectedMock = jest.fn();

  const wrapper = shallow(
    <IntlProviderWrapper>
      <SearchBlocks
        blocks={mockBlocks}
        onSearchFilter={onSearchFilterMock}
        onBlockSelected={onBlockSelectedMock}
      />
    </IntlProviderWrapper>,
  );

  test('renders without crashing', () => {
    expect(wrapper.exists()).toBeTruthy();
  });

  test('snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
