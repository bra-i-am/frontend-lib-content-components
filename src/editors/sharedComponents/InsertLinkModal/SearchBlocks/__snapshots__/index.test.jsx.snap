// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBlocks Component snapshot 1`] = `
<IntlProvider
  defaultFormats={Object {}}
  defaultLocale="en"
  fallbackOnEmptyString={true}
  formats={Object {}}
  locale="en"
  messages={Object {}}
  onError={[Function]}
  onWarn={[Function]}
  textComponent={Symbol(react.fragment)}
>
  <SearchBlocks
    blocks={
      Object {
        "block-children-1": Object {
          "blockId": "edx_block-1",
          "displayName": "Block children 1",
          "id": "block-children-1",
          "legacyWebUrl": "http://localhost/legacy",
          "lmsWebUrl": "http://localhost/weburl",
          "path": "Any display name / Block children 1",
          "studentViewUrl": "http://localhost/studentview",
          "type": "sequential",
        },
        "block-children-2": Object {
          "blockId": "edx_block-2",
          "displayName": "Block children 2",
          "id": "block-children-2",
          "legacyWebUrl": "http://localhost/legacy",
          "lmsWebUrl": "http://localhost/weburl",
          "path": "Any display name / Block children 2",
          "studentViewUrl": "http://localhost/studentview",
          "type": "sequential",
        },
        "block-key": Object {
          "blockId": "edx_block-1",
          "children": Array [
            "block-children-1",
            "block-children-2",
          ],
          "displayName": "Any display name",
          "id": "block-key",
          "legacyWebUrl": "http://localhost/legacy",
          "lmsWebUrl": "http://localhost/weburl",
          "path": "Any display name",
          "studentViewUrl": "http://localhost/studentview",
          "type": "sequential",
        },
      }
    }
    onBlockSelected={[MockFunction]}
    onSearchFilter={[MockFunction]}
    searchInputValue=""
  />
</IntlProvider>
`;
